boards/arm/nrf9160_pca20035/CMakeLists.txt

```diff
@@ -4,12 +4,11 @@
 #
 # SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
 
-zephyr_library()
+if(CONFIG_TRUSTED_EXECUTION_SECURE AND NOT DEFINED CONFIG_MCUBOOT)
+	zephyr_library()
+	zephyr_library_sources(board_secure.c)
+endif()
 
 zephyr_library_sources_ifdef(
 	CONFIG_TRUSTED_EXECUTION_NONSECURE
 	board_nonsecure.c)
```

Comment:
If I understand you correctly, then you should change

zephyr_library_sources_ifdef(
  CONFIG_TRUSTED_EXECUTION_NONSECURE

into

if(CONFIG_TRUSTED_EXECUTION_NONSECURE)

and put a second zephyr_library() invocation into that if-statement (before the "sources" invocation). 

This is to ensure that when CONFIG_TRUSTED_EXECUTION_SECURE, we don't add board_secure.c to some unrelated library that has been created earlier.

Reply:
Thanks! I updated according to your suggestion, and it seems to be working well